pylib/Tools/Executor/sequential.py

```diff
@@ -66,16 +66,24 @@ def execute(self, testDef):
         status = 0
         only_reporter = False
         stageOrder = testDef.loader.stageOrder
+
+        # Holding a semaphore while in transition between plugins
+        # so async threads don't interrupt in the wrong context
+        testDef.plugin_trans_sem.acquire()
```

Comment:
Hi @rhc54 -- I added this comment to describe the semaphores more. Does this help?